googletest/CMakeLists.txt

```diff
@@ -105,15 +105,23 @@ target_link_libraries(gtest_main gtest)
 # to the targets for when we are part of a parent build (ie being pulled
 # in via add_subdirectory() rather than being a standalone build).
 if (DEFINED CMAKE_VERSION AND NOT "${CMAKE_VERSION}" VERSION_LESS "2.8.11")
-  target_include_directories(gtest      SYSTEM INTERFACE "${gtest_SOURCE_DIR}/include")
-  target_include_directories(gtest_main SYSTEM INTERFACE "${gtest_SOURCE_DIR}/include")
+  set(gtest_include_dirs "${gtest_SOURCE_DIR}/include")
+  if (NOT "${CMAKE_VERSION}" VERSION_LESS "3.0.0")
+    set(gtest_include_dirs
+      $<BUILD_INTERFACE:${gtest_include_dirs}>
+      $<INSTALL_INTERFACE:include>)
+  endif()
+  target_include_directories(gtest      SYSTEM INTERFACE ${gtest_include_dirs})
+  target_include_directories(gtest_main SYSTEM INTERFACE ${gtest_include_dirs})
 endif()
 
+
 ########################################################################
 #
 # Install rules
 if(INSTALL_GTEST)
```

Comment:
I do miss the installation of a `GtestConfigVersion.cmake` file for version information. Please add:
```
  include(CMakePackageConfigHelpers)
  write_basic_package_version_file(
    "${CMAKE_CURRENT_BINARY_DIR}/GtestConfigVersion.cmake"
    COMPATIBILITY AnyNewerVersion)
```